### PR TITLE
Fixed a few bugs in dns message str parsing

### PR DESCRIPTION
**Pull request by dvladi77**

### CODE REVIEW

File: libmy/b64_encode.c [hunk 1]
```diff
@@ -73,11 +73,6 @@ int base64_encode_block(const char* plaintext_in, int length_in, char* code_out,
 			*codechar++ = base64_encode_value(result);
 			
 			++(state_in->stepcount);
-			if (state_in->stepcount == CHARS_PER_LINE/4)
-			{
-				*codechar++ = ' ';
-				state_in->stepcount = 0;
```
Comment (shw700) [inline libmy/b64_encode.c:79]: It looks like all instances of this stepcount variable may be removed

File: libmy/b64_encode.c [hunk 2]
```diff
@@ -73,11 +73,6 @@ int base64_encode_block(const char* plaintext_in, int length_in, char* code_out,
 			*codechar++ = base64_encode_value(result);
 			
 			++(state_in->stepcount);
-			if (state_in->stepcount == CHARS_PER_LINE/4)
-			{
-				*codechar++ = ' ';
```
Comment (shw700) [inline libmy/b64_encode.c:78]: Also note that this change breaks the unit tests (which can be run with "make check") as there are still base64 encoded strings in t/test-rdata.c that contain spaces.

File: wdns/str_to_rdata_ubuf.c [hunk 3]
```diff
@@ -1069,7 +1069,7 @@ _wdns_str_to_rdata_ubuf(ubuf *u, const char *str,
 				my_rrtype = wdns_str_to_rrtype(s_rrtype);
 				free(s_rrtype);
 
-				if (my_rrtype == 0 || (rrtype >= 128 && rrtype < 256) || rrtype == 65535) {
```
Comment (cmikk) [inline wdns/str_to_rdata_ubuf.c:1073]: The removal of this rrtype constraint should be noted in the commit message, as it's a change in approach.

The constraints would be appropriate for a DNS server where including these reserved / query-only types would be an application-level error, but _our_ priority is faithfully displaying bitmaps we see in the wild.

(Normally, I'd say such changes in approach should be reflected in comments, but in this case that would be a comment on removed code, which can be confusing.)

File: wdns/str_to_rdata_ubuf.c [hunk 4]
```diff
@@ -1043,7 +1042,8 @@ _wdns_str_to_rdata_ubuf(ubuf *u, const char *str,
 			const char *end;
 			char *s_rrtype;
 			u16buf *rrtypes;
-			uint16_t my_rrtype, last_rrtype;
+			uint16_t my_rrtype;
+			uint32_t last_rrtype;		// last_rrtype has to be bigger that my_rrtype to cover all uint16_t range
```
Comment (shw700) [inline wdns/str_to_rdata_ubuf.c:1046]: *than

Comment (cmikk) [inline wdns/str_to_rdata_ubuf.c:1046]: Also style: please use C-style /* */ comment here.

**Pull request closed without merging**